pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -379,6 +379,19 @@ func (o *MustGatherOptions) Run() error {
 		go func(pod *corev1.Pod) {
 			defer wg.Done()
 
+			if len(o.RunNamespace) > 0 {
+				defer func() {
```

Comment:
In case the `oc adm must-gather` command is interrupted by a user/hanging in the background, the must-gather pod will get removed prematurely/not get removed at all.

Reply:
> In case the `oc adm must-gather` command is interrupted by a user/hanging in the background, the must-gather pod will get removed prematurely/not get removed at all.
> 
> There's also a request to implement `oc adm must-gather [resume]` which will "re-join" the already running must-gather pod in case the command was stopped/interrupted.

Isn't this problem also valid for default must-gather?

Reply:
It is. The internet connection from the client to the server may be interrupted. So the pod will not get deleted. Also, when a temporary namespace is created it will not get removed either.

Even with checking the keep option, the must-gather pod can be still deleted prematurely. Worth checking whether aborting the command will still call the defer here.

Reply:
> It is. The internet connection from the client to the server may be interrupted. So the pod will not get deleted. Also, when a temporary namespace is created it will not get removed either.
> 
> Even with checking the keep option, the must-gather pod can be still deleted prematurely. Worth checking whether aborting the command will still call the defer here.

I checked from the code and if user passes `--keep` flag, I could not find any code portion that can delete namespace or the pod. But for the `--keep=false` case, you are right, namespace or pod become stale after manual interruption. But maybe we can propose a general fix for all cases when user interrupts the connection in a followup PR, since this PR only focuses on removal of pod when user passes `--run-namespace`. What do you think?

Reply:
@ingvagabund what do you think about this https://github.com/openshift/oc/pull/1320#discussion_r1065667621?.

Reply:
Using --keep make sense. As a user I was not aware of this option. So I presume the oc adm must-gather does not remove a pod when the command is accidentally interrupted by a user. Seen an unwritten rule by observation. Nevertheless, given this is a "new" option, we can state the interruption case in the release notes and advise users to consider usage of the keep option.

Reply:
Thanks, sounds good to me